Hermes/appscale/hermes/stats/handlers.py

```diff
@@ -0,0 +1,92 @@
+import json
+import logging
+
+from tornado.web import RequestHandler
+from tornado.options import options
+
+from appscale.hermes.constants import SECRET_HEADER
+
+
+class Respond404Handler(RequestHandler):
+  def initialize(self, reason):
+    self.reason = reason
+
+  def get(self):
+    self.set_status(404, self.reason)
+
+
+class CachedStatsHandler(RequestHandler):
+  """ Handler for reading node/processes/proxies stats history
+  """
+
+  def initialize(self, stats_cache):
+    self._stats_cache = stats_cache
+
+  def get(self):
+    if self.request.headers.get(SECRET_HEADER) != options.secret:
+      logging.warn("Received bad secret from {client}"
+                   .format(client=self.request.remote_ip))
+      self.set_status(401, "Bad secret")
```

Comment:
Thanks, I didn't notice that. And also it makes sense to change 401 (Not authorized) to 403 (Denied).